packages/cozy-sharing/src/components/ShareAutosuggest.jsx

```diff
@@ -49,81 +45,71 @@ export default class ShareAutocomplete extends Component {
         )
   }
 
-  onSuggestionsFetchRequested = ({ value }) => {
-    this.setState(state => ({
-      ...state,
-      suggestions: this.computeSuggestions(value)
-    }))
+  const onSuggestionsFetchRequested = ({ value }) => {
+    setSuggestions(computeSuggestions(value))
   }
 
-  onSuggestionsClearRequested = () => {
-    this.setState(state => ({ ...state, suggestions: [] }))
+  const onSuggestionsClearRequested = () => {
+    setSuggestions([])
   }
-  onPaste = () => {
-    this.isPasted = true
+
+  const onAutosuggestPaste = () => {
+    setIsPasted(true)
   }
 
-  onChange = (event, { newValue, method }) => {
-    if (this.isPasted) {
+  const onAutosuggestChange = (event, { newValue, method }) => {
+    if (isPasted) {
       const emails = extractEmails(newValue)
       if (emails) {
         emails.map(email => {
-          this.onPick({ email })
+          onAutosuggestPick({ email })
         })
       }
-      this.isPasted = false
+      setIsPasted(false)
     } else {
       if (typeof newValue !== 'object') {
-        this.setState(state => ({ ...state, inputValue: newValue }))
+        setInputValue(newValue)
       } else if (method === 'click' || method === 'enter') {
         // A suggestion has been picked
-        this.onPick(newValue)
+        onAutosuggestPick(newValue)
       }
     }
   }
 
-  onKeyPress = event => {
+  const onKeyPress = event => {
     // The user wants to add an unknown email
     if (
       ((event.key === 'Enter' || event.keyCode === 13) &&
-        validateEmail(this.state.inputValue)) ||
+        validateEmail(inputValue)) ||
       ((event.key === 'Space' || event.keyCode === 32) &&
-        validateEmail(this.state.inputValue))
+        validateEmail(inputValue))
     ) {
-      this.onPick({ email: this.state.inputValue })
+      onAutosuggestPick({ email: inputValue })
     }
   }
 
-  onFocus = () => {
-    this.props.onFocus()
+  const onAutosuggestFocus = () => {
+    onFocus()
   }
 
-  onBlur = (event, { highlightedSuggestion }) => {
+  const onAutosuggestBlur = (event, { highlightedSuggestion }) => {
     if (highlightedSuggestion) {
-      this.props.onPick(highlightedSuggestion)
-      this.setState(state => ({ ...state, inputValue: '' }))
-    } else if (
-      this.state.inputValue !== '' &&
-      this.state.inputValue.match(/\S+@\S+/)
-    ) {
-      this.props.onPick({ email: this.state.inputValue })
-      this.setState(state => ({ ...state, inputValue: '' }))
+      onAutosuggestPick(highlightedSuggestion)
+    } else if (inputValue !== '' && inputValue.match(/\S+@\S+/)) {
+      onAutosuggestPick({ email: inputValue })
     }
   }
 
-  onPick = value => {
-    this.props.onPick(value)
-    this.setState(state => ({ ...state, inputValue: '' }))
-    requestAnimationFrame(() => this.input && this.input.focus(), 1) // don't ask...
```

Comment:
If someone can explain me what was the purpose of this line ? I did not see a difference without it (first added here https://github.com/cozy/cozy-drive/commit/c7f95d9d68a66042c87e5acc3cb91e54b9f9e6a4#diff-deb41b822a10ac8f8f645674a1188a96a8ad503e9bf192c09d63c462d94480bcR53)

Reply:
It was about the `requestAnimationFrame` line. The `// don't ask` comment was already before the "CI purpose" commit. CI still passes so it is a good start.

Reply:
In general when you see a `setTimeout(..., 1)` in modern frameworks code, this is used to execute some code on the next rendering loop, so you are sure that every props have been propagated and are up to date. Maybe this is the case here?
Is the selection cursor at the same place with or without this line of code after the event?

Reply:
Thank you for your answer!

- I understand better the setTimeout x next rendering loop thing
- I discovered one case (and maybe there are more) where it was useful to focus (when you remove a recipient, focus stays on the input)